custom/icds/location_reassignment/tests/test_parser.py

```diff
@@ -94,14 +95,37 @@ def test_parser(self, location_type_mock, _):
             workbook = get_workbook(file)
             parser = Parser(self.domain, workbook)
             errors = parser.parse()
-            self.assertEqual(parser.valid_transitions['awc']['Move'], {'131': '112'})
-            self.assertEqual(parser.valid_transitions['awc']['Merge'], {'132': ['113']})
-            self.assertEqual(parser.valid_transitions['supervisor']['Move'], {'13': '12'})
+            self.assertEqual(len(parser.valid_transitions['awc']), 1)
+            awc_transition = attr.asdict(parser.valid_transitions['awc'][0])
+            self.assertEqual(
+                awc_transition,
+                {'domain': self.domain,
+                 'location_type_code': 'awc',
+                 'operation': 'Move',
+                 'old_site_codes': ['112'],
+                 'new_site_codes': ['131'],
+                 'new_location_details': {
+                     '131': {'name': 'AWC 3', 'parent_site_code': '13', 'lgd_code': 'AWC-131'}},
+                 'user_transitions': {'username2': 'username3'}}
+            )
+            self.assertEqual(len(parser.valid_transitions['supervisor']), 1)
+            supervisor_transition = attr.asdict(parser.valid_transitions['supervisor'][0])
```

Comment:
just going by the documentation for attr which suggests using `asdict`

Reply:
Not saying that it is better than implementing the `__dict__` method. Since the library already had the way to do it, used that itself.

Reply:
Broader question: what are you getting out of using the `attr` library? This was the only code that I recognized as coming from that library, but I'm guessing there's more.

Reply:
`attr` lets me use `Transition` model as a class with some nice things like inbuilt validation for [mandatory things](https://github.com/dimagi/commcare-hq/pull/27310/commits/3af755b34c3da762ea2e8b499cf377cc3ed9913d#diff-9ff50b05910fedc69e20552211aff7bcR30-R32) and provide default for others.

`Transition` model is used at multiple places. `attr` is directly only used for the `asdict` call otherwise.